lib/authorizer/index.js

```diff
@@ -104,14 +105,16 @@ authorizeRequest = function (request, done) {
 
     var clonedReq = new sdk.Request(request.toJSON()),
         auth = clonedReq.auth,
+        authInterface = createAuthInterface(auth),
         authVarlist = auth.parameters(), // @todo: wrap in auth interface
         handler;
 
     if (authVarlist) {
         handler = AuthLoader.getHandler(auth.type);
     }
 
-    handler && handler.sign(auth, clonedReq, function () { done(null, clonedReq); });
+    handler && handler.sign(authInterface, clonedReq, function () { return done(null, clonedReq); });
+    // @todo what happens if handler is not found? why there is no return here??
```

Comment:
Maybe let's add `if (!handler) { done('runtime~authorizeRequest: could not find handler for auth type' + auth.type); } `?